tests/sender/send_lookup.py

```diff
@@ -243,6 +243,31 @@ def test_process_fields_does_not_modify_arguments(self):
         self.assertEqual(fields, ["a", "b", "c"])
         self.assertEqual(processed_fields, '"b","a","c"')
 
+    # Add new line with double quotes to lookup
+    def test_ssl_lookup_new_line_with_double_quotes(self):
+        engine_config = SenderConfigSSL(
+            address=(self.server, self.port),
+            key=self.key,
+            cert=self.cert,
+            chain=self.chain,
+            check_hostname=False,
+            verify_mode=CERT_NONE,
+        )
+        con = Sender(engine_config)
+
+        lookup = Lookup(name=self.lookup_name, historic_tag=None, con=con, escape_quotes=True)
+        p_headers = Lookup.list_to_headers(["KEY", "Value"], "KEY")
+        lookup.send_control("START", p_headers, "INC")
+        if len(con.socket.recv(1000)) == 0:
+            raise Exception("Not msg sent!")
+        lookup.send_data_line(key="11", fields=["11", 'If double quotes are not escaped the send fails"'])
+        if len(con.socket.recv(1000)) == 0:
+            raise Exception("Not msg sent!")
+        lookup.send_control("END", p_headers, "INC")
+        if len(con.socket.recv(1000)) == 0:
+            raise Exception("Not msg sent!")
+
+        con.socket.shutdown(0)
```

Comment:
Unfortunately, this test will not prove that the double-quote escaping is properly working. For it, I propose to test the following:

1. `Lookup.clean_field`: let's test this method's output depending on the `field` and `escape_quotes` parameters:
  i. `field` is a string without double quotes and `escape_quotes=False`.
  ii. `field` is a string without double quotes and `escape_quotes=True`.
  iii. `field` is a string with double quotes and `escape_quotes=False`.
  iv. `field` is a string with double quotes and `escape_quotes=True`.
2. `Lookup` constructor: we need to check that the `escape_quotes` attribute gets propagated to the static methods. To do so, implement a [spy](https://wesmckinney.com/blog/spying-with-python-mocks/) over the static methods that we need to ensure the parameter was propagated and check that the essential operations that use these static methods call them correctly (these are `send_data_line` with and without `key_index` parameter set and `send_csv` with and without `delete_index` parameter set). This will ensure you will not have missed any parameter propagation as the one I pointed out for the `Lookup.process_fields` call in the `send_csv` method before.